### PR TITLE
Punishments#checkLockExpiration: Permalock check

### PR DESCRIPTION
**Pull request by panpawn**

### CODE REVIEW

File: punishments.js [hunk 1]
```diff
@@ -1098,6 +1098,7 @@ Punishments.checkLockExpiration = function (userid) {
 	const punishment = Punishments.userids.get(userid);
 
 	if (punishment) {
+		if (Users(userid) && Users(userid).permalocked) return ` (never expires; you are permalocked)`;
```
Comment (Zarel) [inline punishments.js]: Do not repeatedly use accessors like `Users(userid)`. If you need it more than once, make a variable for it.

**Comment by panpawn** [comment]
Updated

File: punishments.js [hunk 2]
```diff
@@ -1098,6 +1098,8 @@ Punishments.checkLockExpiration = function (userid) {
 	const punishment = Punishments.userids.get(userid);
 
 	if (punishment) {
+		let userObj = Users(userid);
```
Comment (Zarel) [inline punishments.js]: Why are you calling this variable `userObj` and not `user`.

What would you expect a variable named `user` to contain, if not a user object?

**Comment by Zarel** [comment]
From now on, we should probably get used to using `const` instead of `let` for variables that aren't modified.

**Pull request merged**